aim/web/ui/src/components/kit_v2/Tree/Tree.tsx

```diff
@@ -0,0 +1,214 @@
+    '&::before': {
+      content: '""',
+      position: 'absolute',
+      zIndex: 10,
+      width: '100%',
+      height: '100%',
+      left: 0,
+    },
+  },
+  '.ant-tree-checkbox-indeterminate': {
+    '.ant-tree-checkbox-inner': {
+      bs: 'inset 0 0 0 1px $colors$primary100',
+    },
+    '.ant-tree-checkbox-inner::after': {
+      top: '50%',
+      left: '50%',
+      size: '6px',
+      br: '$1',
+      bc: '$primary100',
+      border: 0,
+      transform: 'translate(-50%, -50%) scale(1)',
+      opacity: 1,
+      content: ' ',
+    },
+  },
+  '.ant-tree-checkbox-checked': {
+    '&:after': {
+      display: 'none',
+    },
+    '.ant-tree-checkbox-inner': {
+      bc: '$primary100',
+      bs: 'inset 0 0 0 1px $colors$primary100',
+      '&:after': {
+        borderWidth: 1,
+        transition: 'unset',
+        width: '3px',
+        height: '5px',
+      },
+    },
+  },
+  '.ant-tree-checkbox-inner': {
+    size: '10px',
+    border: 'unset',
+    bs: 'inset 0 0 0 1px $colors$secondary100',
+  },
+});
+
+type DataNode = {
+  key: string;
+  title?: React.ReactNode | ((data: DataNode) => React.ReactNode);
+  children?: DataNode[];
+};
+
+const getParentKey = (key: React.Key, tree: DataNode[]): React.Key => {
+  let parentKey: React.Key;
+  for (let i = 0; i < tree.length; i++) {
+    const node = tree[i];
+    if (node.children) {
+      if (node.children.some((item) => item.key === key)) {
+        parentKey = node.key;
+      } else if (getParentKey(key, node.children)) {
+        parentKey = getParentKey(key, node.children);
+      }
+    }
+  }
+  return parentKey!;
+};
+
+const TreeList = ({ searchValue = '', data, ...props }: ITreeProps) => {
+  const [expandedKeys, setExpandedKeys] = React.useState<React.Key[]>([]);
+  const [autoExpandParent, setAutoExpandParent] = React.useState(true);
+
+  const dataList = React.useMemo(() => {
+    const list: DataNode[] = [];
+    const generateList = (nodes: DataNode[]) => {
+      for (let i = 0; i < nodes.length; i++) {
+        const node = nodes[i];
+        const { key } = node;
+        list.push({ key, title: key });
+        if (node.children) {
+          generateList(node.children);
+        }
+      }
+    };
+    generateList(data);
+    return list;
+  }, [data]);
+
+  const onExpand = (newExpandedKeys: React.Key[]) => {
+    setExpandedKeys(newExpandedKeys);
+    setAutoExpandParent(false);
+  };
+
+  React.useEffect(() => {
+    if (searchValue) {
+      const newExpandedKeys = dataList
+        .map((item) => {
+          if (item.key.indexOf(searchValue) > -1) {
+            return getParentKey(item.key, data);
+          }
+          return null;
+        })
+        .filter((item, i, self) => item && self.indexOf(item) === i);
+      setExpandedKeys(newExpandedKeys as React.Key[]);
+      setAutoExpandParent(true);
+    }
+  }, [data, dataList, searchValue]);
+
+  const treeData = React.useMemo(() => {
+    const loop = (d: DataNode[]): DataNode[] =>
```

Comment:
let's change the name of the `loop` function name to more descriptive name. 
Maybe `processTreeData` or `setTreeDataProps`.